flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -442,8 +453,10 @@ public static void runInteractiveCli(YarnClusterClient yarnCluster, boolean read
 						case "quit":
 						case "stop":
 							yarnCluster.shutdownCluster();
+							if (yarnCluster.hasBeenShutdown()) {
+								removeAppState(applicationId);
```

Comment:
Cleanup is handled in the shutdown method. Please remove this. The cluster client is always in state running if the user calls stop.